content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Simplify Your Embedded Projects with ESP-BSP"
+date: 2024-06-18
+showAuthor: false
+authors:
+    - "juraj-michalek"
+tags: ["Embedded Systems", "ESP32", "ESP32-S3", "Espressif", "BSP"]
+---
+
+# Simplify Your Embedded Projects with ESP-BSP
+
+## Introduction
+
+Are you a maker or an embedded systems enthusiast looking to create applications that work seamlessly across different ESP32 development boards? Whether you’re using the [ESP-WROVER-KIT](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/hw-reference/esp32/get-started-wrover-kit.html), [M5Stack-CoreS3](https://docs.m5stack.com/en/core/CoreS3), [ESP32-S3-BOX-3](https://github.com/espressif/esp-box/blob/master/docs/hardware_overview/esp32_s3_box_3/hardware_overview_for_box_3.md), or other compatible boards, the ESP Board Support Package (ESP-BSP) makes your life easier. In this article, we’ll walk you through how to get started with ESP-BSP, enabling you to focus on your project’s functionality without worrying about hardware differences.
+
+## What is ESP-BSP?
+
+[ESP-BSP](https://github.com/espressif/esp-bsp) is a collection of Board Support Packages featuring Espressif’s or M5Stack development boards. The set can be extended to any board with Espressif’s chip. It provides a convenient way to start a project for a specific development board without the need to manually look for drivers and other details. By using ESP-BSP, you can:
+
+- **Streamline Hardware Integration**: Simplify code and reduce complexity.
+- **Enhance Development Speed**: Quickly set up and start working on your projects.
+- **Access Standardized APIs**: Ensure consistency across your projects.
+
+## Getting Started with ESP-BSP
+
+The following example covers steps for building application for ESP32-S3-BOX-3 which is supported by ESP-BSP.
+
+### Hardware Setup
+
+Ensure you have the following hardware:
+
+- [ESP32-S3-BOX-3](https://github.com/espressif/esp-box) development board.
+- USB-C Cable for power and programming.
+
+### Prerequisites
+
+Before you begin, make sure you have the following:
+
+- [ESP-IDF v5.3](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/index.html#installation): The official development framework for the ESP32, properly installed and sourced in your shell.
+
+### Creating Project from Example
+
+Let’s create a simple project using the `display_audio_photo` example, which is available for the ESP32-S3-BOX-3. This example showcases how to use the display, touch, and audio features.
+
+1. **Initialize a New Project**:
+
+   Use the `idf.py` tool to create a new project from the example:
+
+   ```bash
+   idf.py create-project-from-example "espressif/esp-box-3^1.2.0:display_audio_photo"
```

Comment:
Should we mention that the BSP can be added to a project directly?
i.e:

```bash
idf.py add-dependency "espressif/esp_bsp_generic^1.2.0"
```

Reply:
That actually brings an interesting idea. If we would have remove-dependency, we would have a seqequence of commands for switching BSP from CLI.

Reply:
There is add-dependency for the correction of I2C.
I would explain construction of example from the scratch in a different article. This is quick bootstrapping from existing example code.